### PR TITLE
posting a comment and error handling tests

### PR DESCRIPTION
**Pull request by tarynCodes**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -234,3 +234,41 @@ describe("Get /api/articles", ()=>{
...
+    test("POST 404: respond with a 404 when the article id is valid but non exsistant and therefore cannot post comment", () => {
+      return request(app)
+      .post("/api/articles/200000000/comments")
+      .send({body:"cats have nine lives", username: "butter_bridge", article_id: 2})
+      .expect(404)
+      .then((response) => {
+        const {msg} = response.body
+        expect(msg).toBe("No article found!")
+      })
+    })
```
Comment (Slyose) [inline __tests__/app.test.js:408]: Great stuff so far! This would still benefit a lot from some more tests....
- What if the submitted username of the new comment doesnt exist in the db? (404)
- An invalid ID "/api/articles/banana/comments" (400)
- What if there are unnecessary properties on the new comment object, are these ignored and the post request made anyway? (201)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -234,3 +234,41 @@ describe("Get /api/articles", ()=>{
...
+    .expect(400)
+    .then((response) => {
+      const {msg} = response.body
+      expect(msg).toBe("Bad Request!")
+    })
+  })
+    test("POST 404: respond with a 404 when the article id is valid but non exsistant and therefore cannot post comment", () => {
+      return request(app)
+      .post("/api/articles/200000000/comments")
+      .send({body:"cats have nine lives", username: "butter_bridge", article_id: 2})
```
Comment (Slyose) [inline __tests__/app.test.js]: not sure why you've got article_id: 2 in this object - doesn't seem to be relevant to the test, since the article id comes from the URL

File: app.js [hunk 3]
```diff
@@ -17,17 +17,29 @@ app.get('/api/articles/:article_id', getArticleByArticleId)
...
+    next(err)
+  }
+})
+
+app.use((err, req, res, next) => {
+  if(err.status && err.msg) {
+    res.status(err.status).send({msg: err.msg})
   }else{
     next(err)
   }
```
Comment (Slyose) [inline app.js]: Well handled errors! Might be worth considering putting these functions in their own file then requiring them in

**Review by Slyose — COMMENTED** [review]
Love the code so far!

There are some additions that would be great to have in your tests before this is ready to merge, please take a look at the feedback, make changes then resubmit a PR

**Comment by tarynCodes** [comment]
I cant seem to get my custom error handler to work with the error code 23503 - error 404 
I would like to write custom messages like "invalid article/invalid user" but can't seem to make it work. In this case as the two tests are the same error code would you use a different parameter from the error to distinguish between them?

**Comment by Harrison365** [comment]
Hey Taryn :) 

Great additions. You've added all of the tests that Modou asked for 👍 

Just a little unsure on where selectUsers is being invoked? Could you clarify form me? Thanks
